usaspending_api/search/filters/postgres/tas.py

```diff
@@ -0,0 +1,112 @@
+from usaspending_api.common.exceptions import InvalidParameterException
+from usaspending_api.accounts.models import TreasuryAppropriationAccount, FederalAccount
+from django.db.models import Q
+from usaspending_api.search.filters.postgres.HierarchicalFilter import HierarchicalFilter, Node
+from usaspending_api.accounts.helpers import TAS_COMPONENT_TO_FIELD_MAPPING
+
+
+class TasCodes(HierarchicalFilter):
+    underscore_name = "tas_codes"
+
+    @classmethod
+    def build_tas_codes_filter(cls, queryset, filter_values):
+        if isinstance(filter_values, list):
+            # This is a legacy usage, and will be dealt with by the other filter
+            return TreasuryAccounts.build_tas_codes_filter(queryset, filter_values)
+        elif isinstance(filter_values, dict):
+            require = filter_values.get("require") or []
+            exclude = filter_values.get("exclude") or []
+        else:
+            raise InvalidParameterException(f"tas_codes must be an array or object")
+
+        return Q(
+            treasury_account_identifiers__overlap=list(
+                cls._query_string(TreasuryAppropriationAccount.objects.all(), require, exclude).values_list(
+                    "treasury_account_identifier", flat=True
+                )
+            )
+        )
+
+    @staticmethod
+    def node(code, positive, positive_naics, negative_naics):
+        return TASNode(code, positive, positive_naics, negative_naics)
+
+
+def string_to_dictionary(string):
+    if len(string.split("-")) == 1:
+        return {"aid": string}
+    elif len(string.split("-")) == 2:
+        return FederalAccount.fa_rendering_label_to_component_dictionary(string)
+    else:
+        return TreasuryAppropriationAccount.tas_rendering_label_to_component_dictionary(string)
+
+
+def search_regex_of(v):
+    if isinstance(v, str):
+        v = string_to_dictionary(v)
+
+    code_lookup = {
```

Comment:
Potentially dumb question: does this code for sub-awards return the same TAS hierarchy using the recent Federal Account bucketing? The Elasticsearch implementation needed to use additional fields to account for this behavior.

Reply:
The Postgres implementation has the advantage of being able to use foreign  keys directly. see line 73.

Reply:
👍